Test_Game.java

```diff
@@ -15,109 +23,137 @@
+				System.exit(0);
+			}        
+		});    
+
+		mainFrame.getContentPane().setBackground(Color.BLACK);
+
+		mainFrame.add(gameDisplay);
+		mainFrame.setVisible(true);
+	}
+
+
+	private void showKeyListenerDemo(){
+
+		worldy = new World(maxRow, maxCol);
+		beardyGuy = new Controller(worldy);
+
+
+
+		mainFrame.addKeyListener(new CustomKeyListener());
+		mainFrame.setVisible(true);  
+
+		while(gameRunning){
+			try {
+				Thread.sleep(50);                 //1000 milliseconds is one second.
 			} catch(InterruptedException ex) {
-			    Thread.currentThread().interrupt();
+				Thread.currentThread().interrupt();
+			}
+			if (beardyGuy.getScoreAndLives()[1]==0){
+				gameRunning=false;
 			}
-		   beardyGuy.moveEverything();
-		   displayMap();
-	   }
-   }
-   
-   
-   
-
-   private void displayMap(){
-	   String output = new String();
-	   char[][] map = worldy.getMap();
-	   
-	   output = "<html><pre>C:\\game &gt;<br>";
-	   
-	   for(int rowCounter=maxRow;rowCounter>-1;rowCounter--){
-		   output = output+String.valueOf(map[rowCounter])+"<br>";
-	   }
-	   output = output+"Score: "+String.valueOf(beardyGuy.getScoreAndLives()[0])
-			   +"<br>Lives: "+String.valueOf(beardyGuy.getScoreAndLives()[1]) + "</pre></html>";
-	   
-	   
-	   
-	   gameDisplay.setText(output);
-	   
-   }
-
-   class CustomKeyListener implements KeyListener{
-      public void keyTyped(KeyEvent e) {
-      }
-
-      public void keyPressed(KeyEvent e) {
-    	  if(e.getKeyCode() == KeyEvent.VK_ESCAPE){
-         	 gameRunning=false;
-    	  }if(e.getKeyCode() == KeyEvent.VK_UP){
-    		  beardyGuy.userInput('u');
-    	  }if(e.getKeyCode() == KeyEvent.VK_DOWN){
-        	  beardyGuy.userInput('d');
-          }if(e.getKeyCode() == KeyEvent.VK_LEFT){
-        	  beardyGuy.userInput('l');
-          }if(e.getKeyCode() == KeyEvent.VK_RIGHT){
-        	  beardyGuy.userInput('r');
-          }if(e.getKeyCode() == KeyEvent.VK_SPACE){
-        	  beardyGuy.shoot();
-          }
-      }
-
-      public void keyReleased(KeyEvent e) {
-      }   
-   }
- 
+			beardyGuy.moveEverything();
+			displayMap();
+		}
+	}
+
+	public void topScore(){
+		int topScore = fileScan.nextInt();
+		int newScore = beardyGuy.getScore();
+		if (topScore < newScore){
+			out.println(newScore);
+		}
+	}
+
+
+	private void displayMap(){
+		String output = new String();
+		char[][] map = worldy.getMap();
+
+		output = "<html><pre>C:\\game &gt;<br>";
+
+		for(int rowCounter=maxRow;rowCounter>-1;rowCounter--){
+			output = output+String.valueOf(map[rowCounter])+"<br>";
+		}
+		output = output+"Score: "+String.valueOf(beardyGuy.getScoreAndLives()[0])
+		+"<br>Lives: "+String.valueOf(beardyGuy.getScoreAndLives()[1]) + "</pre></html>";
+
+
+
+		gameDisplay.setText(output);
+
+	}
+
+	class CustomKeyListener implements KeyListener{
+		public void keyTyped(KeyEvent e) {
+		}
+
+		public void keyPressed(KeyEvent e) {
+			if(e.getKeyCode() == KeyEvent.VK_ESCAPE){
```

Comment:
would split these if statements up more to better show that they're separate and not one massive block